### PR TITLE
5-rxjs

### PR DESCRIPTION
**Pull request by MaksymSelenkov**
https://maksymselenkov.github.io/frontend-course-2021/homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/index.html

### CODE REVIEW

File: homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/index.html [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        <meta name="viewport"
+            content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+        <meta http-equiv="X-UA-Compatible" content="ie=edge">
+        <title>5-rxjs</title>
+        <link rel="stylesheet" href="style.css">
+    </head>
+    <body>
+        <div class="container" data-container>
+            <header class="header" data-header>
+                <span class="header__logo">Logo</span>
```
Comment (DrSmile444) [inline homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/index.html:14]: In general, when you click on a logo, you expect to be redirected to the main page. For this purpose, you need to mark your logo as a link, not a span. I consider using the `a` tag here.

File: homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/index.html [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                    <div class="header__burger-icon"></div>
+                </div>
+                <nav class="header__nav" data-navigation>
+                    <a class="header__link" href="">News</a>
+                    <a class="header__link" href="">Contacts</a>
+                    <a class="header__link" href="">About us</a>
+                </nav>
+                <button type="button" class="header__buy" data-buy>Buy now</button>
+            </header>
+            <main class="main" data-main>
```
Comment (DrSmile444) [inline homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/index.html:27]: You have never used the 'data-main' selector. So it's better to make your code cleaner and remove it

File: homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const burgerEl = document.querySelector('[data-burger]');
+const navigationEl = document.querySelector('[data-navigation]');
+const buyEl = document.querySelector('[data-buy]');
+
+fromEvent(window, 'scroll')
+    .pipe(
+        throttleTime(150),
+        map((e) => e.currentTarget.scrollY),
+        pairwise(),
+        filter((v) => Math.abs(v[1] - v[0]) > 50),
```
Comment (DrSmile444) [inline homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js:16]: [non-blocking] For me, it's better to destruct arguments and give them descriptive names. `previousScroll` and `currentScroll` sounds better, then `v[1]` and `v[0]`

File: homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        map((v) => v[0] < v[1]),
```
Comment (DrSmile444) [inline homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js:17]: same here

File: homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    )
+    .subscribe((v) => {
```
Comment (DrSmile444) [inline homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js:19]: v? What does it mean? It should describe is header hidden. Try to create a better name for this variable, please

File: homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        if (v) {
+            headerEl.classList.add('header--hide');
+        } else {
+            headerEl.classList.remove('header--hide');
+        }
+    });
+
+fromEvent(burgerEl, 'click')
+    .subscribe((v) => {
+        if (v) {
```
Comment (DrSmile444) [inline homeworks/maksym.selenkov_MaksymSelenkov/5-rxjs/app.js:29]: I'd like to say that you have `v` again, but do you really need this `v` and if? 😄
fromEvent returns PointerEvent, which always true, so this if is useless

**Review by DrSmile444 — CHANGES_REQUESTED** [review]

**Comment by MaksymSelenkov** [comment]
Header covers the main text. How to fix it?

**Review by DrSmile444 — DISMISSED** [review]
Well done! Thank you 👍

**Comment by DrSmile444** [comment]
> Header covers the main text. How to fix it?

I think it's better to add additional space between the top of the page in this case. You can do it by using CSS.

**Comment by MaksymSelenkov** [comment]
Thank you for review)

**Review by DrSmile444 — APPROVED** [review]

**Pull request merged**